pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -369,15 +375,34 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 					log.Errorf("error with get about RAID info from devTree: %v", err)
 				} else {
 					curentRaid = zfs.GetZpoolRaidType(vdevs)
+					for _, vdev := range vdevs.Devices {
+						// If this is a RAID or mirror, look at the disks it consists of
+						if vdev.Type == libzfs.VDevTypeMirror || vdev.Type == libzfs.VDevTypeRaidz {
+							for _, disk := range vdev.Devices {
+								rDiskStatus, err := zfs.GetZfsDiskAndStatus(disk)
+								if err == nil {
```

Comment:
Please log.Errorf if you get an error.

Reply:
In this case, we do not need to display errors if we hit, for example, a snapshot or cache.
If the sent data is not a physical disk, this is normal and possible. In this case, if it is not a disk, then we do not add it.

Reply:
Can you please add a comment about this - that we ignore errors since they appear when we have e.g., a snapshot

Reply:
That comment isn't useful because it just repeats what the code says. The needed part is that vdev.Devices might includes snapshots or caches, and those will result in errors which we need to ignore.